### PR TITLE
add session-db plugin to expose getting session db from hero

### PR DESCRIPTION
**Pull request by GlenDC**
Example usage:

```typescript
const socketRecords = (await hero.readSessionDb()).sockets.all();
// ... e.g. store the relevant info from the sockets as TCP connections
```

```typescript
const resources = (await hero.readSessionDb()).resources.all();
// ... e.g. store the relevant info from the sockets as Http Requests
```

### CODE REVIEW

**Comment by soundofspace** [comment]
Some remarks:
- Do you always want to make a new SessionDb when getSessionDb is called, or do it only once the first time the function is called and store it internally?
- Is the  ` needed in your examples or is it a typo?
- Can you also add these examples and some info about what this plugin does to README.md Plugins section.

File: lib/SessionDb.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import type { Hero } from '@ulixee/hero/lib/extendables';
+import { ClientPlugin } from '@ulixee/hero-plugin-utils';
+import type ISessionTraffic from '../interfaces/ISessionDb';
+import SessionDb from '@ulixee/hero-core/dbs/SessionDb';
+const id = 'animations-plugin';
```
Comment (soundofspace) [inline lib/SessionDb.ts]:
```suggestion
const id = 'session-db-plugin';
```

File: lib/SessionDb.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+export class ClientSessionTrafficPlugin extends ClientPlugin {
```
Comment (soundofspace) [inline lib/SessionDb.ts]:
```suggestion
export class ClientSessionDbPlugin extends ClientPlugin {
```

File: lib/SessionDb.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import type { Hero } from '@ulixee/hero/lib/extendables';
+import { ClientPlugin } from '@ulixee/hero-plugin-utils';
+import type ISessionTraffic from '../interfaces/ISessionDb';
+import SessionDb from '@ulixee/hero-core/dbs/SessionDb';
+const id = 'animations-plugin';
```
Comment (GlenDC) [inline lib/SessionDb.ts]:
```suggestion
const id = 'session-db-plugin';
```

File: lib/SessionDb.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+export class ClientSessionTrafficPlugin extends ClientPlugin {
```
Comment (GlenDC) [inline lib/SessionDb.ts]:
```suggestion
export class ClientSessionDbPlugin extends ClientPlugin {
```

File: lib/SessionDb.ts [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    });
+    return sessionDb;
+}
+
+declare module '@ulixee/hero/lib/extendables' {
+    interface Hero extends ISessionTraffic { }
+    interface Tab extends ISessionTraffic { }
+}
+
+export default { ClientSessionTrafficPlugin }
```
Comment (GlenDC) [inline lib/SessionDb.ts]:
```suggestion
export default { ClientSessionDbPlugin }
```

File: lib/SessionDb.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    const sessionId = await hero.sessionId;
+    const sessionDb = new SessionDb(sessionId, {
+        readonly: true,
+        fileMustExist: true,
+    });
+    return sessionDb;
+}
+
+declare module '@ulixee/hero/lib/extendables' {
+    interface Hero extends ISessionTraffic { }
```
Comment (GlenDC) [inline lib/SessionDb.ts]:
```suggestion
    interface Hero extends ISessionDb { }
```

File: lib/SessionDb.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    interface Tab extends ISessionTraffic { }
```
Comment (GlenDC) [inline lib/SessionDb.ts]:
```suggestion
    interface Tab extends ISessionDb { }
```

File: interfaces/ISessionDb.ts [hunk 8]
```diff
@@ -0,0 +1,5 @@
+import SessionDb from '@ulixee/hero-core/dbs/SessionDb';
+
+export default interface ISessionTraffic {
```
Comment (GlenDC) [inline interfaces/ISessionDb.ts]:
```suggestion
export default interface ISessionDb {
```

File: lib/SessionDb.ts [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import type { Hero } from '@ulixee/hero/lib/extendables';
+import { ClientPlugin } from '@ulixee/hero-plugin-utils';
+import type ISessionTraffic from '../interfaces/ISessionDb';
```
Comment (GlenDC) [inline lib/SessionDb.ts]:
```suggestion
import type ISessionDb from '../interfaces/ISessionDb';
```

**Review by soundofspace — APPROVED** [review]

**Pull request merged**